manifests/init.pp

```diff
@@ -29,14 +29,22 @@
     info('puppet_agent performs no actions on Puppet 4+')
   }
   else {
-    class { '::puppet_agent::prepare': } ->
-    class { '::puppet_agent::install': } ->
-    class { '::puppet_agent::config': } ~>
-    class { '::puppet_agent::service': }
+    if $::architecture == 'x86' and $arch == 'x64' {
+      fail('Unable to install x64 on a x86 system')
+    }
+    if $::osfamily == 'windows' {
+      class { '::puppet_agent::osfamily::windows': }
```

Comment:
No critical files need to be moved, and @kylog said we are not supporting moving modules installed via pmt to there new home.  Same with cache folder